exp/api/v1beta1/azuremachinepool_default.go

```diff
@@ -102,3 +139,35 @@ func (amp *AzureMachinePool) SetNetworkInterfacesDefaults() {
 		}
 	}
 }
+
+// findParentMachinePool finds the parent MachinePool for the AzureMachinePool.
+func (amp *AzureMachinePool) findParentMachinePool(cli client.Client) (*expv1.MachinePool, error) {
+	ctx := context.Background()
+	machinePoolList := &expv1.MachinePoolList{}
+	if err := cli.List(ctx, machinePoolList); err != nil {
+		return nil, errors.Wrapf(err, "failed to list MachinePools %s/%s", amp.Namespace, amp.Name)
+	}
+	for _, mp := range machinePoolList.Items {
+		if mp.Spec.Template.Spec.InfrastructureRef.Name == amp.Name {
+			return &mp, nil
+		}
+	}
+	return nil, errors.Errorf("failed to get MachinePool %s/%s", amp.Namespace, amp.Name)
+}
+
+// getSubscriptionID returns the subscription ID for the given AzureMachinePool.
+func (amp *AzureMachinePool) getSubscriptionID(cli client.Client, clusterName string) (string, error) {
```

Comment:
seems like this could also be shared/reused between AzureMachine and MachinePool (we'd just need to pass in the namespace too)